src/test/unit/libs/test_sampling.py

```diff
@@ -1,134 +1,166 @@
 import pytest
+from opentelemetry.sdk.trace.sampling import Decision
+from opentelemetry.trace import SpanKind
 
 from lumigo_opentelemetry.libs.sampling import (
-    _extract_url,
-    _should_skip_span_on_route_match,
+    _extract_endpoint,
+    _get_string_list_from_env_var,
+    does_match_regex_safe,
+    does_endpoint_match_filtering_regexes,
+    AttributeSampler,
 )
 
 
 @pytest.mark.parametrize(
-    "attributes, expected_url",
+    "endpoint, spanKind, matches_server_filter, matched_client_filter, matched_general_filter, expected_sampling_decision",
     [
-        ({"url.full": "http://test.com:80"}, "http://test.com"),
-        ({"url.full": "http://test.com:443"}, "http://test.com:443"),
-        ({"url.full": "https://test.com:443"}, "https://test.com"),
-        ({"url.full": "https://test.com:80"}, "https://test.com:80"),
-        ({"url.full": "http://test.com"}, "http://test.com"),
-        ({"http.url": "http://test.com"}, "http://test.com"),
+        ("/orders", SpanKind.SERVER, True, False, False, Decision.DROP),
+        ("/orders", SpanKind.SERVER, False, True, False, Decision.RECORD_AND_SAMPLE),
+        ("/orders", SpanKind.SERVER, False, False, True, Decision.DROP),
         (
-            {"url.full": "http://test.com/test", "http.route": "/test"},
-            "http://test.com/test",
+            "https://foo.bar",
+            SpanKind.CLIENT,
+            True,
+            False,
+            False,
+            Decision.RECORD_AND_SAMPLE,
         ),
+        ("https://foo.bar", SpanKind.CLIENT, False, True, False, Decision.DROP),
+        ("https://foo.bar", SpanKind.CLIENT, False, False, True, Decision.DROP),
+    ],
+)
+def test_should_sample(
+    endpoint,
+    spanKind,
+    matches_server_filter,
+    matched_client_filter,
+    matched_general_filter,
+    expected_sampling_decision,
+    monkeypatch,
+):
+    monkeypatch.setattr(
+        "lumigo_opentelemetry.libs.sampling._extract_endpoint",
+        lambda *args, **kwargs: endpoint,
+    )
+    monkeypatch.setattr(
+        "lumigo_opentelemetry.libs.sampling.does_endpoint_match_server_filtering_regexes",
+        lambda *args, **kwargs: matches_server_filter,
+    )
+    monkeypatch.setattr(
+        "lumigo_opentelemetry.libs.sampling.does_endpoint_match_client_filtering_regexes",
+        lambda *args, **kwargs: matched_client_filter,
+    )
+    monkeypatch.setattr(
+        "lumigo_opentelemetry.libs.sampling.does_endpoint_match_filtering_regexes",
+        lambda *args, **kwargs: matched_general_filter,
+    )
```

Comment:
TBH, I think that mocking 4 internal functions is a lot to abstract away. 
I guess controlling the result in the `SERVER` kind is harder so we can mock that part, but for the client one (and the "old" env-var for both client and server) maybe we can just craft values that trigger the expected result?

Reply:
We have separate unit tests for the:
* _extract_endpoint func
* does_endpoint_match_filtering_regexes func, which is the base for the three does_endpoint_match_[server\client\]_filtering_regexes funcs

I wanted to test the logic of testing if we have an endpoint, and if the server / client / general regex is matched (And to make sure we are not testing a client endpoint with a server regex for example)